### ISSUE TITLE
Incorrect login credentials message despite "Sign in success" shown in security log

### ISSUE THREAD

**Issue by ewencb** [ProtonMail/proton-bridge#260]
## Expected Behavior
Login with protonmail credentials to the protonmail-bridge app and see bridge connection details.

## Current Behavior
Using the same credentials I use for the web app I get "Incorrect login credentials" error on the bridge application. In the security log on the web client I see "Sign in success" despite not being logged in on the application.

## Version Information
2.1.1

## Context (Environment)
Cannot use any external mail client.

**Comment by sirius-c** [comment]
Same issue here on Xubuntu 21.10 since about yesterday - after entering the correct second-factor code I get "Incorrect login credentials". Can't access Protonmail via the Bridge anymore.

Interestingly, I cannot see the Bridge version without signing in - but I cannot sign in. - Latest version of the Bridge, though.

**Comment by andrzejsza** [comment]
could you both send us logs to bridge@protonmail.com? will look into it right away.

**Comment by sirius-c** [comment]
Logs sent.

**Comment by 03k64** [comment]
I had a similar issue to this (including the successful sign in entries in the logs) and it turned out to be that I did not have a recognised password store running (ie. `pass` or `gnome-keyring`). 

There was no indication of this in the GUI in the latest version of the bridge (`2.1.1`) installed from the AUR. I ran into the error after switching to KDE from Gnome, I assume I must have removed `gnome-keyring` from my system without realising the dependency when I was cleaning up the previous Gnome install.

I ended up downgrading to my previously installed version (`1.8.12`) which _did_ display an error when starting the bridge GUI that indicated what I needed to do. I ended up installing `pass` and setting it up per the [documentation](https://www.passwordstore.org/) and the Bridge is now working again.

**Comment by sirius-c** [comment]
Thanks for that information. I definitely have the GNOME key ring running, but this observation might point to a kind-of race condition at login, if the Bridge comes up faster than the key ring. Maybe faster than with earlier Bridge versions?

**Comment by 03k64** [comment]
Possibly? I'll leave that to the devs to investigate. I received no information from the latest version at either application start or on failed login to point to the error though, so that feels like a regression compared to the previous version. The previous version wouldn't even let me attempt to log in until I'd installed and setup `pass`

**Comment by andrzejsza** [comment]
thanks for the logs, we're still not sure - it does simply look as if the keychain was not accessible.
the only regression to V1 is just about the error message - we are no longer displaying keychain-specific error - we'll fix that in the upcoming releases.
We're also working on allowing general 'secretservice' instead of 'gnome-keyring' - should be ready in the next release too.

**Comment by sirius-c** [comment]
1. Now I can log in again with the seocnd factor. Possibly the issue from a few days back was due to a hick-up on Proton with respect to verifying the second factor?
2. After I log in manually, I can close the Bridge and restart it without being asked login details. These are clearly stored and retrieved form the keyring.
3. When rebooting, I found myself logged out of my Proton account each time.
4. I now added a 10 second delay to the start of the Bridge after booting - there seems to be no more logging out.

So possibly an issue of 'who comes up first'.
